windows/guest_customization/win_gosc_prepare.yml

```diff
@@ -17,47 +17,80 @@
     customize_timezone_name: "Hawaiian Standard Time"
     gosc_dns_servers: ['192.168.0.1', '192.168.0.2']
     customize_runonce_echo_string: 'Windows gosc automation test'
-- name: Set fact of the run once command
+
+- name: "Set fact of the run once command"
   ansible.builtin.set_fact:
     customize_runonce: "cmd.exe /c echo {{ customize_runonce_echo_string }} > C:\\gosc_runonce.txt"
 
-- name: Set fact of the VM guest IP before GOSC
+- name: "Set fact of the VM guest IP before GOSC"
   ansible.builtin.set_fact:
     guest_ip_before_gosc: "{{ vm_guest_ip }}"
 
-# Get guest OS hostname from VM guestinfo
-- include_tasks: ../../common/vm_get_config.yml
+- name: "Get guest OS hostname from VM guestinfo"
+  include_tasks: ../../common/vm_get_config.yml
   vars:
     property_list: ['guest.hostName']
 
-- name: Set fact of the hostname before GOSC
+- name: "Set fact of the hostname before GOSC"
   ansible.builtin.set_fact:
     hostname_before_gosc: "{{ vm_config.guest.hostName }}"
+
 - ansible.builtin.debug:
     msg: "Get guest OS hostname/IP before customization: {{ hostname_before_gosc }}/{{ guest_ip_before_gosc }}"
 
-- name: Set fact of default Windows dir
+- name: "Set fact of default Windows dir"
   ansible.builtin.set_fact:
     win_dir: '$env:windir'
-- include_tasks: ../utils/win_get_path.yml
+
+- name: "Get OS directory"
+  include_tasks: ../utils/win_get_path.yml
   vars:
     win_get_path_specified: "{{ win_dir }}"
-- name: Set fact of the absolute path of Windows dir
+
+- name: "Set fact of the absolute path of Windows dir"
   ansible.builtin.set_fact:
     win_dir: "{{ win_get_path_absolute }}"
 - ansible.builtin.debug:
     msg: "Windows GOSC log files in Windows dir: {{ win_dir }}"
 
 # Uninstall OneDrive in Windows 11 for the known 3rd-party issue
 # Paramter 'uninstall_onedrive' is used for internal testing only
-- include_tasks: uninstall_onedrive.yml
+- name: "Uninstall OneDrive"
+  include_tasks: uninstall_onedrive.yml
   when:
     - uninstall_onedrive is defined
     - uninstall_onedrive | bool
     - guest_os_ansible_distribution_ver is version('10.0.22000.0', '>=')
     - guest_os_product_type | lower == 'client'
 
+- name: "Initialize BitLocker vars"
+  ansible.builtin.set_fact:
+    need_disable_bitlocker: false
```

Comment:
This parameter is not necessary. Below 'when' condition can use service status or volume status directly.

Reply:
Fixed. Thanks. And also remove the block.